setuptools/tests/test_manifest.py

```diff
@@ -602,3 +605,27 @@ def test_prune(self):
         file_list.sort()
         assert file_list.files == ['a.py', ml('f/f.py')]
         self.assertWarnings()
+
+    def test_warning_on_meaningless_parameter_combination(self):
```

Comment:
I don't think we want a test for this behavior. Tests are mainly for behavior we wish to enforce. This particular behavior, that a warning is issued, is incidental to the intention. I'm not sure it quite captures what we need.

Reply:
Are you saying we shouldn't add a test for the warning at all, or that this particular test doesn't correctly cover the use case?

I personally think we *should* be testing that a warning is raised in the right conditions (and, preferably, asserting that *no* warning is raised under the *wrong* conditions as well, though I think #1823 should cover us in that situation, since any uncaptured warning would be considered a failure).

In general I don't love the idea of exact matches on the warning message, though. To me the user-facing behavior we want to enforce is, "A warning will be issued in this situation that describes the problem," so the precise wording of `'Distutils 'package_data'...`. Normally I would avoid false positives by checking the warning *type*, but I do think it's common to use the `self.warn` mechanism here instead of the warnings module.

I think maybe we can do something a bit looser, use a `assert_called_once` and then check that the message includes `package_data` and `include_package_data`, possibly with a comment like, "This is a heuristic to detect whether the correct warning was raised; the exact warning message is not guaranteed to be stable."

Reply:
I agree. This test does provide some value. In fact, it enabled me to readily confirm my suspicions that the behavior wasn't quite what was intended.

My main concern with this test (and its phrasing) is that it seems to indicate that supplying these two parameters is _intended_ to be incompatible, which contradicts the previously-documented intention that these parameters should be compatible. Assuming we find an acceptable implementation, I'll tweak the phrasing and maybe add a docstring to the test to explain how this test is describing behavior and not prescribing behavior.